source/common/router/router.cc

```diff
@@ -2026,6 +2026,9 @@ uint32_t Filter::numRequestsAwaitingHeaders() {
 
 bool Filter::checkDropOverload(Upstream::ThreadLocalCluster& cluster,
                                std::function<void(Http::ResponseHeaderMap&)>& modify_headers) {
+  if (!Runtime::runtimeFeatureEnabled("envoy.reloadable_features.enable_drop_overload")) {
```

Comment:
Please add this as part of the `if (cluster.dropOverload().value())` below.
This will couple the change with the code (so if later new code is added after the if scope, it will be used whether the runtime feature is enabled or not).